### PR TITLE
Add mocha tests for app.connect method

### PR DESCRIPTION
**Pull request by jacquesdev**
@jack-lewin - have added one test to start off, just to get mocha in place, let me know if you are happy with this, happy to make edits where required..

I was hoping to test whether the github object was actually successfully authenticated or not, but I wasn't sure if there is any property on the object to tell me this, I guess it's something useful to test. I did add this issue to check if something like this was available (and maybe you know). https://github.com/mikedeboer/node-github/issues/597.

If you are happy with this, I will hand it over to Theo to add some more of the actual tests.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -21,11 +21,12 @@ exports.connect = function (formattedTime) {
   })
 
   // authenticate with the GitHub API, if possible
-  if (process.env.GITHUB_OAUTH) {
+  if (process.env.GITHUB_OAUTH !== undefined) {
```
Comment (jack-lewin) [inline app.js]: Could we make this `if (typeof process.env.GITHUB_OAUTH !== 'undefined')`?

[This StackOverflow answer explains it well](https://stackoverflow.com/a/4725697) 👍

**Review by jack-lewin — CHANGES_REQUESTED** [review]
Looking good! 👌

One thing I should've mentioned - I'm using [StandardJS](https://standardjs.com) as a style guide/linter. Would you mind just making sure we've got a consistent style, please? It comes as a plugin for most editors, there's a pretty good list on the website. Thanks a lot!

**Comment by jacquesdev** [comment]
Hi @jack-lewin.

I have made the updates as requested, and have added standard.js to the project which means it will run the linter as part of the tests (and should now look ok in your editor).

I have left sinon in there, because ideally I should probably stub the github.authenticate method instead of checking the actual value, but I was having some troubles getting that to work. But will look at updating that after. In the meantime, when this gets merged, Theo can get started.

**Review by jack-lewin — APPROVED** [review]

**Comment by jack-lewin** [comment]
Awesome, thanks for adding standard to `npm test` as well. Looking good for @tmarkovic to get started.

**Comment by tmarkovic** [comment]
Hey! 

Nicely done, I'll try to dig in after work. I see no reason for me to monopolize this issue so if you're willing and able, write tests like there's no tomorow and I'll try to contribute where needed 👍

**Comment by jacquesdev** [comment]
Thanks @tmarkovic - I will try and add a couple of tests nearer the end of the week, so feel free to start where @jack-lewin suggested, basically the module that follows the connect module.

**Pull request merged**